public/tier0/dbg.h

```diff
@@ -566,8 +566,16 @@ class CScopeMsg
 // We're using an ancient version of GCC that can't quite handle some
 // of our complicated templates properly.  Use some preprocessor trickery
 // to workaround this
-#ifdef __GNUC__
-	#define COMPILE_TIME_ASSERT( pred ) typedef int UNIQUE_ID[ (pred) ? 1 : -1 ]
+#ifndef __has_feature
+  #define __has_feature(x) 0 // Compatibility with non-clang compilers.
+#endif
+#if __has_feature(cxx_static_assert)

```

Comment:
FWIW, we only support compilers that have static_assert now.  But preserving the intent of this code is fine.